packages/wonder-stuff-i18n/src/utils/i18n-utils.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import fs from "fs";
+import path from "path";
+
+import ancesdir from "ancesdir";
+
+import {extractStrings} from "./extract-i18n.js";
+
+import type {ExtractedString} from "./extract-i18n.js";
+
+export type TranslatedLocaleStrings = {
+    [msgid: string]: string | {|lang: string, messages: Array<string>|},
+};
+
+/**
+ *
+ * @param {string} ignoreFile The file to read the glob strings from.
+ * @returns {Array<string>} a list of glob strings to ignore
+ */
+export const getIgnoreGlobs = (
+    ignoreFile: string = ".i18nignore",
+): Array<string> => {
+    try {
+        // Find the ignore file, if it exists
+        // If it doesn't exist then ancesdir throws an exception.
+        const ignoreDir = ancesdir(__dirname, ignoreFile);
+
+        const foundIgnoreFile = path.join(ignoreDir, ignoreFile);
+        return (
+            fs
+                .readFileSync(foundIgnoreFile, "utf8")
+                .split("\n")
+                // Remove comments and trailing whitespace
+                .map((line) => line.replace(/\s*#.*$/g, "").trim())
+                // Remove empty strings
+                .filter((line) => !!line)
+        );
+    } catch (e) {
+        // If the file doesn't exist, then we're not ignoring any files.
+        return [];
+    }
+};
+
+/**
+ * Extracts the location of the i18n strings in the given code string.
+ *
+ * @param {string} textString a string of code to extract strings from
+ * @returns {Array<Object>} the strings that were extracted from the given
+ *  code string
+ */
+export const getI18nStringsFromString = (
+    textString: string,
+): Array<ExtractedString> => {
+    const strings = [];
+
+    // Get all the strings that need to be replaced, but in reverse
+    // order, so that it's easier to insert them in the right place.
+    for (const extractedString of extractStrings(textString)) {
+        strings.unshift(extractedString);
+    }
+
+    return strings;
+};
```

Comment:
Now that we have types for things I feel much better having a bunch of different things using `strings` as their identifier. 🙂